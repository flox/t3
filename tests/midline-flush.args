-- sh -c 'for i in `seq 1 10`; do tests/midline-flush $i; sleep 0.01; done'
