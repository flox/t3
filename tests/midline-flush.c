/*
 * command that performs write system call to stdout then calls
 * flush then writes a second line with a newline.
 */

#define _GNU_SOURCE
#include <stdio.h>
#include <unistd.h>

#define hello(fd) fprintf(fd, "Hello, %s! ", fd == stdout ? "stdout" : "stderr")
#define goodbye(fd) fprintf(fd, "Goodbye, %s!\n", fd == stdout ? "stdout" : "stderr")

int main() {
    hello(stdout);
    hello(stderr);
    fflush(stdout);
    fflush(stderr);
    goodbye(stderr);
    goodbye(stdout);
    fflush(stderr);
    usleep(10); // It only takes a millisecond to give stderr a head start
                // but we'll give it ten to avoid false positives where the
                // first line is written to stdout.
    fflush(stdout);
    usleep(10); // Sleep once more to ensure that stdout has a chance to
                // flush its buffer before the subsequent test runs.
    return 0;
}
