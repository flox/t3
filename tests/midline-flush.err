1: Hello, stderr! 1: Goodbye, stderr!
2: Hello, stderr! 2: Goodbye, stderr!
3: Hello, stderr! 3: Goodbye, stderr!
4: Hello, stderr! 4: Goodbye, stderr!
5: Hello, stderr! 5: Goodbye, stderr!
6: Hello, stderr! 6: Goodbye, stderr!
7: Hello, stderr! 7: Goodbye, stderr!
8: Hello, stderr! 8: Goodbye, stderr!
9: Hello, stderr! 9: Goodbye, stderr!
10: Hello, stderr! 10: Goodbye, stderr!
