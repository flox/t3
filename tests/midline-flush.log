[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
[36m[0m[1m[33mHello, stderr! Goodbye, stderr![0m
[36m[0mHello, stdout! Goodbye, stdout![0m
