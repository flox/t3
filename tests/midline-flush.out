1: Hello, stdout! 1: Goodbye, stdout!
2: Hello, stdout! 2: Goodbye, stdout!
3: Hello, stdout! 3: Goodbye, stdout!
4: Hello, stdout! 4: Goodbye, stdout!
5: Hello, stdout! 5: Goodbye, stdout!
6: Hello, stdout! 6: Goodbye, stdout!
7: Hello, stdout! 7: Goodbye, stdout!
8: Hello, stdout! 8: Goodbye, stdout!
9: Hello, stdout! 9: Goodbye, stdout!
10: Hello, stdout! 10: Goodbye, stdout!
